app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -113,10 +113,31 @@ def save_child_inventory(obj, hashes, child_keys, *args)
 
   def store_ids_for_new_records(records, hashes, keys)
     keys = Array(keys)
-    hashes.each do |h|
-      r = records.detect { |r| keys.all? { |k| r.send(k) == r.class.type_for_attribute(k.to_s).cast(h[k]) } }
```

Comment:
@Fryguy so I could keep the casting I think, if I load those upfront. I will need my morning brain for this though. :-)